src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs

```diff
@@ -0,0 +1,1336 @@
+                //             t1 = default; // 1
+                Diagnostic(ErrorCode.WRN_DefaultExpressionMayIntroduceNullT, "default").WithArguments("T").WithLocation(12, 18),
+                // (15,9): warning CS8602: Possible dereference of a null reference.
+                //         t1.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t1").WithLocation(15, 9),
+                // (25,9): warning CS8602: Possible dereference of a null reference.
+                //         t2.ToString(); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t2").WithLocation(25, 9)
+                );
+        }
+
+        [WorkItem(32503, "https://github.com/dotnet/roslyn/issues/32503")]
+        [Fact]
+        public void PatternDeclarationBreaksNullableAnalysis()
+        {
+            var source = @"
+#nullable enable
+class A { }
+class B : A
+{
+    A M()
+    {
+        var s = new A();
+        if (s is B b) {}
+        return s; 
+    }
+} 
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void InferenceWithITuplePattern()
+        {
+            var source = @"
+#nullable enable
+class A { }
+class B : A
+{
+    A M()
+    {
+        var s = new A();
+        if (s is B b) {}
+        return s; 
+    }
+} 
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void RecursivePatternNullInferenceWithDowncast_01()
+        {
+            var source = @"
+#nullable enable
+class Base
+{
+    public object Value = """";
+}
+class Derived : Base
+{
+    public new object Value = """";
+}
+class Program
+{
+    void M(Base? b)
+    {
+        if (b is Derived { Value: null })
+            b.Value.ToString();
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void RecursivePatternNullInferenceWithDowncast_02()
+        {
+            var source = @"
+#nullable enable
+class Base
+{
+    public object Value = """";
+}
+class Derived : Base
+{
+}
+class Program
+{
+    void M(Base? b)
+    {
+        if (b is Derived { Value: null })
+            b.Value.ToString(); // 1
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (15,13): warning CS8602: Possible dereference of a null reference.
+                //             b.Value.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "b.Value").WithLocation(15, 13));
+        }
+
+        [Fact]
+        public void TuplePatternNullInference_01()
+        {
+            var source = @"
+#nullable enable
+class Program
+{
+    void M((object, object) t)
+    {
+        if (t is (1, null))
+        {
+        }
+        else
+        {
+            t.Item2.ToString(); // 1
```

Comment:
Consider also accessing `Item1`

Reply:
This was marked as resolved but I don't see an `Item1` access. Was it added to a different test?

---
In reply to: [267959405](https://github.com/dotnet/roslyn/pull/34273#discussion_r267959405) [](ancestors = 267959405)

Reply:
Ignore me. CodeFlow got stuck on an old iteration of the change where `Item1` wasn't used. It just refreshed and I see it now. 

---
In reply to: [268735633](https://github.com/dotnet/roslyn/pull/34273#discussion_r268735633) [](ancestors = 268735633,267959405)